### ISSUE TITLE
503 deployment issue

### ISSUE THREAD

**Issue by nloui** [zodern/meteor-up#549]
Hey there,

I'm able to run my app locally but for whatever reason, mup says that everything worked but it's responding with a 503 error.

The only funky thing I'm seeing is:

node-pre-gyp ERR! Tried to download(404): https://github.com/kelektiv/node.bcrypt.js/releases/download/v1.0.2/bcrypt_lib-v1.0.2-node-v46-linux-x64.tar.gz  
node-pre-gyp ERR! Pre-built binaries not found for bcrypt@1.0.2 and node@4.8.0 (node-v46 ABI) (falling back to source compile with node-gyp)  

Otherwise:

Started TaskList: Pushing Meteor App
[52.20.89.155] - Pushing Meteor App Bundle to The Server
[52.20.89.155] - Pushing Meteor App Bundle to The Server: SUCCESS

Started TaskList: Configuring App
[52.20.89.155] - Pushing the Startup Script
[52.20.89.155] - Pushing the Startup Script: SUCCESS
[52.20.89.155] - Sending Environment Variables
[52.20.89.155] - Sending Environment Variables: SUCCESS

Started TaskList: Start Meteor
[52.20.89.155] - Start Meteor
[52.20.89.155] - Start Meteor: SUCCESS
[52.20.89.155] - Verifying Deployment
[52.20.89.155] - Verifying Deployment: SUCCESS

**Comment by zodern** [comment]
Are you using custom certificates or let's encrypt?

**Comment by tomtom87** [comment]
I am getting a 503 on AWS, made a ticket here https://github.com/zodern/meteor-up/issues/565

**Comment by bennygenel** [comment]
@zodern I'm having the same issue with letsencrypt

**Comment by tomtom87** [comment]
seems its briked for everyone with ssl otherwise its working fine

**Comment by zodern** [comment]
I just setup ssl for one of my projects, and it worked perfectly.

@bennygenel and @nloui, could you please run these commands on the server and post the output here or in a gist? Also, please post your config.

```
docker ps
docker exec <app name>-nginx-proxy cat /etc/nginx/conf.d/default.conf
docker inspect --format '{{ .NetworkSettings.IPAddress }}' <app name>
```

**Comment by bennygenel** [comment]
@zodern I'm using load balancer. thats why my root url and ssl url is different.
my config 

```
    servers: {
        one: {
            host: 'host.ip.ip.host',
            username: 'ubuntu',
        },
    },

    meteor: {
        name: 'appname',
        path: '../',
        servers: {
            one: {},
            //two:{}
        },
        buildOptions: {
            serverOnly: true,
        },
        env: {
            ROOT_URL: 'https://app.appname.co',
            MONGO_URL: 'mongodb://username:pass@url/meteor',
            MONGO_OPLOG_URL: 'mongodb://username:pass@url/local?authSource=admin&replicaSet=repset0',
        },

        ssl: {
            // Enables let's encrypt (optional)
            autogenerate: {
                email: 'email@email.com',
                domains: 'be1.appname.co' 
            }
        },
        dockerImage: 'abernix/meteord:base',
        deployCheckWaitTime: 60,
        enableUploadProgressBar: false
    }
```

```
$ docker ps
CONTAINER ID        IMAGE                                    COMMAND                  CREATED              STATUS              PORTS                                      NAMES
60a737942128        jrcs/letsencrypt-nginx-proxy-companion   "/bin/bash /app/en..."   24 seconds ago       Up 23 seconds                                                  appname-nginx-letsencrypt
fc8b41033407        jwilder/nginx-proxy                      "/app/docker-entry..."   39 seconds ago       Up 39 seconds       0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   appname-nginx-proxy
667ffeb98a0f        abernix/meteord:base                     "/bin/sh -c 'bash ..."   About a minute ago   Up About a minute   80/tcp                                     appname


--------------------------

$ docker exec appname-nginx-proxy cat /etc/nginx/conf.d/default.conf
# If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
# scheme used to connect to this server
map $http_x_forwarded_proto $proxy_x_forwarded_proto {
  default $http_x_forwarded_proto;
  ''      $scheme;
}
# If we receive X-Forwarded-Port, pass it through; otherwise, pass along the
# server port the client connected to
map $http_x_forwarded_port $proxy_x_forwarded_port {
  default $http_x_forwarded_port;
  ''      $server_port;
}
# If we receive Upgrade, set Connection to "upgrade"; otherwise, delete any
# Connection header that may have been passed to this server
map $http_upgrade $proxy_connection {
  default upgrade;
  '' close;
}
# Set appropriate X-Forwarded-Ssl header
map $scheme $proxy_x_forwarded_ssl {
  default off;
  https on;
}
gzip_types text/plain text/css application/javascript application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
log_format vhost '$host $remote_addr - $remote_user [$time_local] '
                 '"$request" $status $body_bytes_sent '
                 '"$http_referer" "$http_user_agent"';
access_log off;
# HTTP 1.1 support
proxy_http_version 1.1;
proxy_buffering off;
proxy_set_header Host $http_host;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $proxy_connection;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
proxy_set_header X-Forwarded-Ssl $proxy_x_forwarded_ssl;
proxy_set_header X-Forwarded-Port $proxy_x_forwarded_port;
# Mitigate httpoxy attack (see README for details)
proxy_set_header Proxy "";
server {
	server_name _; # This is just an invalid value which will never trigger on a real hostname.
	listen 80;
	access_log /var/log/nginx/access.log vhost;
	return 503;
}
# be1.appname.co
upstream be1.appname.co {
				## Can be connect with "bridge" network
			# appname
			server 172.17.0.2:80;
}
server {
	server_name be1.appname.co;
	listen 80 ;
	access_log /var/log/nginx/access.log vhost;
	return 301 https://$host$request_uri;
}
server {
	server_name be1.appname.co;
	listen 443 ssl http2 ;
	access_log /var/log/nginx/access.log vhost;
	ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
	ssl_ciphers 'ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS';
	ssl_prefer_server_ciphers on;
	ssl_session_timeout 5m;
	ssl_session_cache shared:SSL:50m;
	ssl_session_tickets off;
	ssl_certificate /etc/nginx/certs/be1.appname.co.crt;
	ssl_certificate_key /etc/nginx/certs/be1.appname.co.key;
	ssl_dhparam /etc/nginx/certs/be1.appname.co.dhparam.pem;
	add_header Strict-Transport-Security "max-age=31536000";
	include /etc/nginx/vhost.d/default;
	location / {
		proxy_pass http://be1.appname.co;
	}
}


------------------------

$ docker inspect --format '{{ .NetworkSettings.IPAddress }}' appname
172.17.0.2
```

**Comment by zodern** [comment]
From what you showed me everything seems to be running correctly.

You can look at the nginx logs with `docker logs <app name>-nginx-proxy`.

Did using the load balancer with let's encrypt work in the past?

**Comment by bennygenel** [comment]
@zodern yes the configuration was fine. I just redeployed the app with the same configuration. meteor didn't give any errors but I think it was because of one of my replica set was down. I started it again and it started working. Also I downgrade to previous version of mup too but don't know if it solvled the problem or not. I'll try tomorrow with the new version again.

Thank you for your time

**Comment by tomtom87** [comment]
I have the 503 error with load balancer and no SSL certificate (because its on AWS and the load balancer has the SSL on it) so I just need HTTPS to be enabled. I made a ticket about it https://github.com/zodern/meteor-up/issues/565

**Comment by zodern** [comment]
There are two problems:

1) Receiving the 503 nginx error page makes it think the app successfully started. The next version will include a fix.
2) The reason it returns the 503 error usually is from let's encrypt unable to verify the domain. You can view the logs using  `docker logs <app-name>-nginx-letsencrypt` and `docker logs <app-name>-nginx-proxy` to try to locate the cause.

**Comment by tomtom87** [comment]
@zodern right, I see! Looking forward to the fix :)

**Comment by zodern** [comment]
I published a 1.2.7 beta that fixed a few bugs with let's encrypt. You can install it with `npm i -g mup@next`.

**Comment by zodern** [comment]
1.2.7 has been released for a month. If you still have problems, feel free to reopen this issue.
